x-pack/plugins/security_solution/public/cases/components/case_settings/sync_alerts_switch.test.tsx

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { mount } from 'enzyme';
+import { act } from '@testing-library/react';
+
+import { SyncAlertsSwitch } from './sync_alerts_switch';
+
+describe('SyncAlertsSwitch', () => {
+  it('it renders', async () => {
+    const wrapper = mount(<SyncAlertsSwitch disabled={false} />);
+
+    expect(wrapper.find(`[data-test-subj="sync-alerts-switch"]`).exists()).toBeTruthy();
+  });
+
+  it('it toggles the switch', async () => {
+    const wrapper = mount(<SyncAlertsSwitch disabled={false} />);
+
+    act(() => {
```

Comment:
You should use `waitFor` here instead. See: https://kentcdodds.com/blog/common-mistakes-with-react-testing-library. You can also get rid of the wrapper.update. that article says do the side effect, then wrap the expect in `waitFor`:

```
  it('it toggles the switch', async () => {
    const wrapper = mount(<SyncAlertsSwitch disabled={false} />);

    wrapper.find('button[data-test-subj="sync-alerts-switch"]').first().simulate('click');
    await waitFor(() => {
      expect(wrapper.find('[data-test-subj="sync-alerts-switch"]').first().prop('checked')).toBe(
        false
      );
    });
  });
```

Reply:
Thanks for the article!